client/ayon_core/plugins/loader/delete_old_versions.py

```diff
@@ -0,0 +1,390 @@
+        versions_to_keep: int,
+        remove_publish_folder: bool,
+        entity_ids: set[str],
+        selection: LoaderActionSelection,
+    ):
+        versions_by_product_id = collections.defaultdict(list)
+        for version in selection.entities.get_products_versions(entity_ids):
+            # Keep hero version
+            if versions_to_keep != 0 and version["version"] < 0:
+                continue
+            versions_by_product_id[version["productId"]].append(version)
+
+        versions_to_delete = []
+        for product_id, versions in versions_by_product_id.items():
+            if versions_to_keep == 0:
+                versions_to_delete.extend(versions)
+                continue
+
+            if len(versions) <= versions_to_keep:
+                continue
+
+            versions.sort(key=lambda v: v["version"])
+            for _ in range(versions_to_keep):
+                if not versions:
+                    break
+                versions.pop(-1)
+            versions_to_delete.extend(versions)
+
+        self.log.debug(
+            f"Collected versions to delete ({len(versions_to_delete)})"
+        )
+
+        version_ids = {
+            version["id"]
+            for version in versions_to_delete
+        }
+        if not version_ids:
+            return LoaderActionResult(
+                message="Skipping. Nothing to delete.",
+                success=False,
+            )
+
+        project = selection.entities.get_project()
+        anatomy = Anatomy(project["name"], project_entity=project)
+
+        repres = selection.entities.get_versions_representations(version_ids)
+
+        self.log.debug(
+            f"Collected representations to remove ({len(repres)})"
+        )
+
+        filepaths_by_repre_id = {}
+        repre_ids_by_version_id = {
+            version_id: []
+            for version_id in version_ids
+        }
+        for repre in repres:
+            repre_ids_by_version_id[repre["versionId"]].append(repre["id"])
+            filepaths_by_repre_id[repre["id"]] = [
+                anatomy.fill_root(repre_file["path"])
+                for repre_file in repre["files"]
+            ]
+
+        size = 0
+        for filepaths in filepaths_by_repre_id.values():
+            for filepath in filepaths:
+                if os.path.exists(filepath):
+                    size += os.path.getsize(filepath)
+
+        if identifier == "calculate-versions-size":
+            return LoaderActionResult(
+                message="Calculated size",
+                success=True,
+                form=LoaderActionForm(
+                    title="Calculated versions size",
+                    fields=[
+                        UILabelDef(
+                            f"Total size of files: {format_file_size(size)}"
+                        ),
+                    ],
+                    submit_label=None,
+                    cancel_label="Close",
+                ),
+            )
```

Comment:
I'm unsure myself about abstracting this away into custom Form classes, etc. What benefit do we really get from enforcing user prompts to be pushed out as a return value here? Why can't this just be a custom Qt dialog? I feel like as soon as we get to only being able to use strict Attribute Defs we're limiting what an action could essentially show as a UI to the user... and most devs will know Qt in our industry, but _everyone_ would need to read and learn this 'custom form designing API' we're introducing as  a return value here. Also, what if I want the form to respond dynamically, etc.